jmclient/jmclient/wallet.py

```diff
@@ -1032,6 +999,499 @@ def close(self):
     def __del__(self):
         self.close()
 
+class PSBTWalletMixin(object):
+    """
+    Mixin for BaseWallet to provide BIP174
+    functions.
+    """
+    def __init__(self, storage, **kwargs):
+        super(PSBTWalletMixin, self).__init__(storage, **kwargs)
+
+    @staticmethod
+    def get_fee_from_psbt(in_psbt):
+        assert isinstance(in_psbt, btc.PartiallySignedTransaction)
+        spent = sum(in_psbt.get_input_amounts())
+        paid = sum((x.nValue for x in in_psbt.unsigned_tx.vout))
```

Comment:
I wonder if it is a good idea to add `get_output_amounts()` method that would do this. Seems that it could improve readability of the code

Reply:
Well perhaps; but on the other hand, the output amounts are directly present, whereas it's a bit more complex in the general case, for inputs, right. I don't mind.

Reply:
It is a reasonable convenience method. I have added it in https://github.com/Simplexum/python-bitcointx/commit/476dcfa18ff494f3983ea51099ddf72e797309ad

Reply:
Speaking of convenience, `get_fee()` is also a reasonable convenience method. I added it in https://github.com/Simplexum/python-bitcointx/commit/7785292109728999fa33bc1eef321d5fca5ed22b

It will raise ValueError if the calculated fee happen to be negative.

Reply:
> It will raise ValueError if the calculated fee happen to be negative.

unless `allow_negative=True` kwarg is passed to `get_fee()` (this was added in later commit)

Reply:
Cool. Will use that, thanks.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/e9367fd755bfbb94c5bb1b73a07a2550536b942c